lib/keygen/version.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require_relative 'jsonapi/renderer'
+require_relative 'jsonapi/errors'
+
+module Keygen
+  VERSION = '1.5.0'.freeze
+
+  module Version
+    class << self
+      def revision
+        return @revision if defined? @revision
+
+        @revision = from_environment || from_git || from_file
+      end
+
+      def version
+        return @version if defined? @version
+
+        @version = Gem::Version.new(VERSION)
+      end
+
+      private
+
+      def from_environment = ENV['COMMIT_HASH'] || ENV['COMMIT_SHA'] || ENV['HEROKU_SLUG_COMMIT']
+      def from_git         = `git rev-parse HEAD` rescue nil
+      def from_file(path = Rails.root / 'VERSION')
+        return unless File.exist?(path)
+
+        File.read(path).chomp
```

Comment:
This should filter out `$Id$` in case it wasn't replaced with `ident` by git.

Reply:
This is the resulting `VERSION` file whenever the repo is cloned:

```
$Id: 055c8729cdcc372500a08db659c045e16c4409fb $
```

So a regexp would work best here.